app/services/mentors/create.rb

```diff
@@ -67,5 +61,26 @@ def mentor_attributes
         pupil_premium_uplift: pupil_premium_uplift?(start_year),
       }
     end
+
+    def user
+      # NOTE: This will not update the full_name if the user has an active participant profile,
+      # the scenario I am working on is enabling a NPQ user to be added as a mentor
+      # Not matching on full_name means this works more smoothly for the end user
+      # and they don't get "email already in use" errors if they spell the name differently
+      @user ||= User.find_or_create_by!(email:) do |mentor|
+        mentor.full_name = full_name
+      end
+    end
+
+    def teacher_profile
+      @teacher_profile ||= TeacherProfile.find_or_create_by!(user:) do |profile|
+        profile.school = school_cohort.school
+      end
+    end
+    alias_method :create_teacher_profile, :teacher_profile
+
+    def participant_profile_exists?
+      teacher_profile.participant_profiles.mentors.exists? || user.participant_identities.joins(:participant_profiles).where(participant_profiles: { type: "ParticipantProfile::Mentor" }).exists?
```

Comment:
I have a feeling we do this elsewhere too. Think it's worth having a `Participants::AlreadyExists` service?

Reply:
@peteryates I'd be happy to implement such service, I couldn't find where something similar was being done.
But I'll keep that one mind.